lib/src/main/java/space/taran/arklib/domain/kind/LinkKindFactory.kt

```diff
@@ -0,0 +1,46 @@
+package space.taran.arklib.domain.kind
+
+import kotlinx.serialization.Serializable
+import space.taran.arklib.ResourceId
+import space.taran.arklib.domain.index.ResourceMeta
+import space.taran.arklib.domain.meta.MetadataStorage
+import java.nio.file.Path
+
+object LinkKindFactory : ResourceKindFactory<ResourceKind.Link> {
+    override val acceptedExtensions = setOf("link")
+    override val acceptedKindCode = KindCode.LINK
+    override val acceptedMimeTypes: Set<String>
+        get() = setOf()
+
+    override fun fromPath(
+        path: Path,
+        meta: ResourceMeta,
+        metadataStorage: MetadataStorage
+    ): ResourceKind.Link {
+        // TODO: we don't need these lines anymore, but what about the native bindings?
```

Comment:
Not relevant to this PR... but @mdrlzy maybe you know if we need to do something like this in our PRs to ARK Shelf?